polygeist/tools/cgeist/Test/Verification/compoundassign.c

```diff
@@ -0,0 +1,174 @@
+// RUN: cgeist %s --function=* -S -o - | FileCheck %s
+
+// CHECK-LABEL:   func.func @f1(
+// CHECK-SAME:                  %[[VAL_0:.*]]: memref<?xi32>,
+// CHECK-SAME:                  %[[VAL_1:.*]]: i32) -> i32
+// CHECK:           %[[VAL_2:.*]] = affine.load %[[VAL_0]][0] : memref<?xi32>
+// CHECK:           %[[VAL_3:.*]] = arith.remsi %[[VAL_2]], %[[VAL_1]] : i32
+// CHECK:           affine.store %[[VAL_3]], %[[VAL_0]][0] : memref<?xi32>
+// CHECK:           return %[[VAL_3]] : i32
+// CHECK:         }
+
+int f1(int *a, int b) {
+  return *a %= b;
+}
+
+// CHECK-LABEL:   func.func @f2(
+// CHECK-SAME:                  %[[VAL_0:.*]]: memref<?xf32>,
+// CHECK-SAME:                  %[[VAL_1:.*]]: i64,
+// CHECK-SAME:                  %[[VAL_2:.*]]: f64) -> f32
+// CHECK:           %[[VAL_3:.*]] = arith.constant 2.000000e+00 : f64
+// CHECK:           %[[VAL_4:.*]] = arith.subf %[[VAL_2]], %[[VAL_3]] : f64
+// CHECK:           %[[VAL_5:.*]] = arith.index_cast %[[VAL_1]] : i64 to index
+// CHECK:           %[[VAL_6:.*]] = affine.load %[[VAL_0]][symbol(%[[VAL_5]])] : memref<?xf32>
+// CHECK:           %[[VAL_7:.*]] = arith.extf %[[VAL_6]] : f32 to f64
+// CHECK:           %[[VAL_8:.*]] = arith.subf %[[VAL_7]], %[[VAL_4]] : f64
+// CHECK:           %[[VAL_9:.*]] = arith.truncf %[[VAL_8]] : f64 to f32
+// CHECK:           affine.store %[[VAL_9]], %[[VAL_0]][symbol(%[[VAL_5]])] : memref<?xf32>
+// CHECK:           return %[[VAL_9]] : f32
+// CHECK:         }
```

Comment:
Differs from old code: it now extends to `f64`, operates there and casts back to `f32` to store and return. It operated in `f32` before.